### PR TITLE
Save layout

### PR DESCRIPTION
**Pull request by dextercd**
Hey! I'm pretty happy with the Pythonification of the save layout. I'm not the best Python programmer, so if you see anything unusual or think the code could be more straightforward please do give feedback on that.

The library exposes the field types and the list of field descriptors. The field descriptors are grouped into categories. The dictionary keys are meant to be used to select a specific descriptor and should generally not change so that programs using the library won't break.

`command_line.py` is a small example program. It can be useful when figuring out fields in the save file, so it is also exposed as a console script named s2save-to-text.

### CODE REVIEW

**Review by gmjosack — COMMENTED** [review]
Hey this is awesome. I'll take a look tonight

File: src/s2_data/field_descriptors.py [hunk 1]
```diff
@@ -0,0 +1,685 @@
+from dataclasses import dataclass
+from enum import IntFlag, IntEnum
+import typing
+from .field_types import *
+
+class CharacterUnlockFlags(IntFlag):
```
Comment (gmjosack) [inline src/s2_data/field_descriptors.py:6]: Just personal preference but I tend to find shifts easier to read for correctness

```py
    Ana        = 1 << 0
    Margaret   = 1 << 1
    Colin      = 1 << 2
    Roffy      = 1 << 3
    Alto       = 1 << 4
    Liz        = 1 << 5
    Nekka      = 1 << 6
```
etc.

**Review by gmjosack — APPROVED** [review]
Other than the one nit I think this looks great. I'll merge in for now and we can iterate from there rather than on this PR if we want to make any changes.

**Comment by dextercd** [comment]
Great!  I'll make that change and also add a few tests

**Pull request merged**